adb-vagrantfile-tests/k8s-tests/test-k8s.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+
+import subprocess
+import unittest
+import re
+import os
+from unittest import TestCase
+
+
+regex_for_vagrantfile = re.compile(
+    "components/centos/centos-k8s-singlenode-setup/Vagrantfile"
+)
+
+
+def is_k8s_custom_vagrantfile_modified():
+    """Check if any custom vagrantfile is modified."""
+    modified_files = git_diff_tree()
+
+    for _ in modified_files:
+        if regex_for_vagrantfile.match(_):
+            return True
+        else:
+            continue
+    return False
+
+
+def git_diff_tree():
+    """Return a list of files that got modified in the PR."""
+    files = subprocess.check_output(['git', 'diff-tree', '--no-commit-id',
+                                     '--name-only', '-r', "%s" %
+                                     (os.environ['ghprbActualCommit'])],
+                                    cwd="/root/adb")
+    return files.split("\n")
+
+
+def configure_git():
+    # To be able to checkout the PR branch
+    subprocess.check_call(
+        ["git",
+         "config",
+         "--add",
+         "remote.origin.fetch",
+         "+refs/pull/*/head:refs/remotes/origin/pr/*"
+         ],
+        cwd="/root/adb"
+    )
+
+    # Fetch all branches (including PRs)
+    subprocess.check_call(
+        ["git", "fetch", "origin"],
+        cwd="/root/adb"
+    )
+
+    # Checkout the PR
+    subprocess.check_call(
+        ["git", "checkout", "pr/%s" % os.environ["ghprbPullId"]],
+        cwd="/root/adb"
+    )
+
+
+class KubernetesTests(TestCase):
+
+    """This class tests Kubernetes on ADB box."""
+
+    def test_01_service_manager_install(self):
+        """Install required plugins."""
+        # vagrant-service-manager plugin is required to boot up ADB box
+        subprocess.check_call(
+            ['vagrant', 'plugin', 'install', 'vagrant-service-manager']
+        )
+
+    def test_02_VagrantUp(self):
+        """Check if vagrant up succeeds."""
+        try:
+            # ADB repo is cloned to /root/adb using Ansible
+            exit_code = subprocess.check_call(
+                ['vagrant', 'up'],
+                cwd='/root/adb/components/centos/centos-k8s-singlenode-setup'
+            )
+        except:
+            pass
+        self.assertEqual(exit_code, 0)
+
+    def test_03_VagrantSSH(self):
+        """Check if k8s is properly setup."""
+        # Dirty hack to ensure below check doesn't fail due to service taking
+        # time to start
+        subprocess.call(["sleep", "5"])
+        try:
+            output = subprocess.check_output(
+                ['vagrant', 'ssh', '-c', '%s' % "kubectl get nodes"],
+                cwd="/root/adb/components/centos/centos-k8s-singlenode-setup"
+            )
+        except:
+            pass
+        self.assertIn('127.0.0.1', output.split())
+        self.assertIn('Ready', output.split())
```

Comment:
I think I should rename this something specific to Kubernetes check instead of checking Vagrant SSH. That's what it's doing anyway.